### PR TITLE
Core/Misc: replace ACE based typedefs

### LINKED ISSUE #2459: Server crash via SQL error
The player tried to withdraw money from the bank. And the server crashed.

**Server_2019-11-19_02-34-27.log**
2019-11-19 20:00:01 ERROR: Error while parsing SQL. Core fix required.

**SQLdriver.log**
```
2019-11-19 20:00:01 SQL: An error occurred formatting string "INSERT INTO log_money VALUES(%u, %u, "%s", "%s", %u, "%s", %u, "<GB WITHDRAW> %s (guild id: %u, members: %u, new amount: %I64u, leader guid low: %u, char level: %u)", NOW())" : invalid type specifier
2019-11-19 20:00:01 ERROR: [1064] You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'An error occurred formatting string "INSERT INTO log_money VALUES(%u, %u, "%s", ' at line 1
```

##### STEPS TO REPRODUCE THE PROBLEM:
1. Open Guild-Bank
2. Withdraw > 10 Gold
3. SQL error, server shutdown.

##### EXTRA NOTES:
SThe SQL statement at Guild.cpp never changed and worked fine except for this point.

##### BRANCH(ES):
master ded3bd8900e1+

##### OPERATING SYSTEM:
 Windows 7/10, MySQL server ver: 5.7.21

SQL-table has old records made by this statement, and then it was recorded fine:
(ip censored)
81	807	Antony 192.***.***.118	0		1000000	<GB WITHDRAW> Ultra Guild (guild id: 19, members: 11, new amount: 13218292, leader guid low: 807, char level: 80)	2019-11-11 14:29:58

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/84049034-server-crash-via-sql-error?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Viste**
##### CHANGES PROPOSED:
- Replace ACE based typedefs

##### ISSUES:
- Close #2459

##### TODO:
- [x] Support x32 platform
- [x] Fix build in Win

##### Target branch(es):
- [x] Master

### CODE REVIEW

**Comment by Winfidonarleyan** [comment]
This PR resolve issues #2459

File: src/common/Define.h [hunk 1]
```diff
@@ -9,10 +9,10 @@
 
 #include <cstdint>
 #include <cstddef>
+#include <climits>
 #include <sys/types.h>
```
Comment (Winfidonarleyan) [inline src/common/Define.h]:
```cpp
#include <cstddef>
```

Comment (Viste) [reply src/common/Define.h]: ?#include <cstdint>
#include <cstddef>
#include <climits>

File: src/server/scripts/Commands/cs_debug.cpp [hunk 2]
```diff
@@ -731,14 +731,14 @@ class debug_commandscript : public CommandScript
...
-                    handler->PSendSysMessage("queue(" SIZEFMTD "): For the item with guid %d, the owner's guid (%d) and the player's guid (%d) don't match!", i, item->GetGUIDLow(), GUID_LOPART(item->GetOwnerGUID()), player->GetGUIDLow());
+                    handler->PSendSysMessage("queue(%zu): For the item with guid %d, the owner's guid (%d) and the player's guid (%d) don't match!", i, item->GetGUIDLow(), GUID_LOPART(item->GetOwnerGUID()), player->GetGUIDLow());
                     error = true;
                     continue;
                 }
 
                 if (item->GetQueuePos() != i)
                 {
-                    handler->PSendSysMessage("queue(" SIZEFMTD "): For the item with guid %d, the queuepos doesn't match it's position in the queue!", i, item->GetGUIDLow());
+                    handler->PSendSysMessage("queue(%zu): For the item with guid %d, the queuepos doesn't match it's position in the queue!", i, item->GetGUIDLow());
```
[the remark this answers is not in the document]
Comment (Viste) [reply src/server/scripts/Commands/cs_debug.cpp]: yes true :\

File: src/common/Define.h (hunk 1 above)
Comment (Winfidonarleyan) [reply src/common/Define.h]: Yes, see TC this file

Comment (Viste) [reply src/common/Define.h]: good plz ) Андрей! Она там есть!!!!

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan is this ready to be merged? If yes, please update the label

**Comment by Viste** [comment]
> @Winfidonarleyan is this ready to be merged? If yes, please update the label

no i need change %zu or win x32 get fucked again

File: src/common/Define.h (hunk 1 above)
Comment (Winfidonarleyan) [reply src/common/Define.h]:
```
#include "CompilerDefs.h"
#include <cstddef>
#include <cinttypes>
#include <climits>
```
All includes in this file

**Comment by Winfidonarleyan** [comment]
```cpp
#include "CompilerDefs.h"
#include <cstddef>
#include <cinttypes>
#include <climits>
```
All includes in this file, need delete other

**Comment by Winfidonarleyan** [comment]
Okay, need test this in win and in linux

File: src/server/scripts/Commands/cs_debug.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Viste) [reply src/server/scripts/Commands/cs_debug.cpp]: all should work with SZFMTD "%" PRIuPTR and %zu

[the remark this answers is not in the document]
Comment (Viste) [reply src/server/scripts/Commands/cs_debug.cpp]: negative effect can be only from objectmgr se last update

**Comment by Viste** [comment]
ready

File: src/server/game/Events/GameEventMgr.cpp [hunk 3]
```diff
@@ -1270,7 +1270,7 @@ void GameEventMgr::GameEventSpawn(int16 event_id)
 
     if (internal_event_id < 0 || internal_event_id >= int32(mGameEventCreatureGuids.size()))
     {
-        sLog->outError("GameEventMgr::GameEventSpawn attempt access to out of range mGameEventCreatureGuids element %i (size: " SIZEFMTD ")",
+        sLog->outError("GameEventMgr::GameEventSpawn attempt access to out of range mGameEventCreatureGuids element %i (size: %zu)",
```
Comment (Winfidonarleyan) [inline src/server/game/Events/GameEventMgr.cpp]: need rework all `%zu`

Comment (Viste) [reply src/server/game/Events/GameEventMgr.cpp]: The PRIuPTR macro (from <inttypes.h>) defines a decimal format for uintptr_t, which should always be large enough that you can cast a size_t to it without truncating

Comment (Viste) [reply src/server/game/Events/GameEventMgr.cpp]: доебал

Comment (Winfidonarleyan) [reply src/server/game/Events/GameEventMgr.cpp]: ха

**Comment by Winfidonarleyan** [comment]
Okay, perfect work, can merge

**Pull request merged**